src/librustc/infer/error_reporting.rs

```diff
@@ -555,6 +555,155 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         }
     }
 
+    fn highlight_outer(&self,
```

Comment:
meta-comment: I've been talking to @cengizIO about making this `error_reporting` section into a proper module, as I think it's starting to have a lot of code. It feels like this type-diffing stuff really wants to be its own module.